### PR TITLE
Status notification

### PR DESCRIPTION
**Pull request by Matthew-Moles-Civica**

### CODE REVIEW

File: src/services/status.service.js [hunk 1]
```diff
@@ -1,15 +1,49 @@
 const {
```
Comment (michalrak) [inline src/services/status.service.js]: This module needs JsDoc and logging

File: src/services/status.service.test.js [hunk 2]
```diff
@@ -58,29 +62,129 @@ describe("status.service getStatus()", () => {
...
 
     it("should return the new value of the status name", () => {
       expect(result).toBe("new value");
     });
+
+    it("should call mocks with the correct values to send an email", () => {
+      expect(getEmailDistribution).toHaveBeenCalledTimes(1);
+      expect(sendSingleEmail).toHaveBeenCalledTimes(1);
+      expect(sendSingleEmail).toHaveBeenLastCalledWith(
+        "e6692529-52a3-45cb-96b7-0c8cfe282167",
```
Comment (michalrak) [inline src/services/status.service.test.js]: If this value is not set in .env, this and two other tests fail. Greater isolation is required in unit tests.

File: src/services/status.service.js (hunk 1 above)
Comment (thomas-finch) [reply src/services/status.service.js]: Added documentation and logging

File: src/services/status.service.js [hunk 3]
```diff
@@ -1,15 +1,49 @@
...
+        status_value: statusText,
+        time: new Date().toLocaleString('en-GB', { hour12: false, timeZone: 'Europe/London'})
+    };
+    
+    const emailList = await getEmailDistribution();
+
+    emailList.forEach(function (item) {
+      try {
+        sendSingleEmail(NOTIFY_STATUS_TEMPLATE, item, data);
+      } catch (err) {
```
[the remark this answers is not in the document]
Comment (thomas-finch) [reply src/services/status.service.js]: Included error handling

File: src/services/status.service.js [hunk 4]
```diff
@@ -1,15 +1,49 @@
...
   const updatedStatusValue = await updateStoredStatus(statusName, newStatus);
+
+  if(newStatus !== currentValue) {
+    let statusText = "Invalid";
+    if(newStatus === true) {
+      statusText = "Succeeded";
+    } else if(newStatus === false) {
+      statusText = "Failed";
+    }
+    statusName = statusName.replace('Succeeded','').replace('mostRecent','').replace(/([A-Z])/g, ' $1').toLowerCase().trim();
```
[the remark this answers is not in the document]
Comment (thomas-finch) [reply src/services/status.service.js]: Added comment

File: src/services/status.service.test.js (hunk 2 above)
Comment (thomas-finch) [reply src/services/status.service.test.js]: Fixed unit tests to not rely on this value

File: src/services/status.service.js [hunk 5]
```diff
@@ -1,28 +1,152 @@
...
+    }
+
+    /**
+     * Format status name to be more readable for use in email
+     * Replace function steps:
+     * 1. Replace first instance of 'Succeeded' with empty string
+     * 2. Replace first instance of 'mostRecent' with empty string
+     * 3. Replace all instances of capital letters with itself prefixed with a space (e.g. 'newStatusName' => 'new Status Name')
+     */
+    let formattedStatusName = statusName.replace('Succeeded','').replace('mostRecent','').replace(/([A-Z])/g, ' $1').toLowerCase().trim();
```
Comment (michalrak) [inline src/services/status.service.js]: Thanks for deciphering this, wouldn't it be worth having a private function for formatting this string?

File: src/services/status.service.js [hunk 6]
```diff
@@ -1,28 +1,152 @@
...
+    const data = {
+        environment_description: ENVIRONMENT_DESCRIPTION,
+        status_name: formattedStatusName.charAt(0).toUpperCase() + formattedStatusName.slice(1),
```
Comment (michalrak) [inline src/services/status.service.js]: A function for transforming this string with some JsDoc explanation would be nice

File: src/connectors/status/status-db.connector.js [hunk 7]
```diff
@@ -10,10 +10,11 @@ const { logEmitter } = require("../../services/logging.service");
 let client;
 let statusDB;
 let statusCollection;
+let emailCollection;
 
 /**
  * Sets up a connection to the status collection in the config database.
- * The client, configDB and statusCollection variables are accessible to other functions in this connector.
+ * The client, configDBemailCollection,  and statusCollection variables are accessible to other functions in this connector.
```
Comment (michalrak) [inline src/connectors/status/status-db.connector.js]: Just a missing comma here

File: src/services/status.service.js [hunk 8]
```diff
@@ -1,30 +1,145 @@
...
+      status_value: statusText,
+      time: new Date().toLocaleString("en-GB", {
+        hour12: false,
+        timeZone: "Europe/London"
+      })
+    };
+
+    const emailList = await getEmailDistribution();
+
+    emailList.forEach(function(item) {
```
Comment (michalrak) [inline src/services/status.service.js]: Please use es6, also can we use a better variable name than "item"?

File: src/services/status.service.js [hunk 9]
```diff
@@ -1,30 +1,145 @@
...
+
   return updatedStatusValue;
 };
 
+/**
+ * Function that increments the count value for a status
+ *
+ * @param {string} statusName The status field name
+ *
+ * @returns {any} The new status value
```
Comment (michalrak) [inline src/services/status.service.js]: Is it returning "any" or a string? Or an integer?

File: src/services/status.service.js [hunk 10]
```diff
@@ -1,30 +1,145 @@
...
+  logEmitter.emit("functionSuccess", "status.service", "getStatus");
+
   return statusName ? status[statusName] : status;
 };
 
+/**
+ * Updates a specified status value
+ *
+ * @param {string} statusName The status field name
+ * @param {any} newStatus The new status value
```
Comment (michalrak) [inline src/services/status.service.js]: Can you set this to specific type

File: src/services/status.service.js [hunk 11]
```diff
@@ -1,30 +1,145 @@
...
+ *
+ * @returns {any} The new status value
```
Comment (michalrak) [inline src/services/status.service.js]: this too

File: src/services/status.service.js (hunk 10 above)
Comment (thomas-finch) [reply src/services/status.service.js]: This could potentially be any type (e.g. Int, Bool, Double)

File: src/services/status.service.js (hunk 11 above)
Comment (thomas-finch) [reply src/services/status.service.js]: This could also potentially be any type (e.g. Int, Bool, Double)

**Pull request merged**